oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -84,6 +85,14 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         provider.augmentProcessParameters(command, classPath);
         // TODO: is this the best way to add our JAR?
         classPath.add(System.getProperty("java.class.path"));
+
+        // Attempt to locate our JAR
+        final ArchiveLocator locator = new ArchiveLocator();
+        final String myClassPath = locator.locateArchive(UserAgentImpl.class);
+        if (myClassPath != null) {

```

Comment:
I don't fully understand all the inner workings of classloaders, so if the getResource() call returns null, we will fail to locate the archive (yeah it's a pretty bad name, I originally only thought of locating jars/zips, but unit test quickly showed I must take care about folders too.)  

The "java.class.path" value is a safety net, what if user passed the JavaFX jar in the classpath?